AlfrescoApp/Model/Managers/RealmSyncManager.m

```diff
@@ -205,23 +205,81 @@ - (void)disableSyncForAccountFromConfig:(UserAccount *)account
-            if(encounteredObstacle)
+    if(node)
+    {
+        __weak typeof(self) weakSelf = self;
+        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+            RLMRealm *backgroundRealm = [RLMRealm defaultRealm];
+            NSMutableArray *arrayOfNodesToDelete = [NSMutableArray new];
+            NSMutableArray *arrayOfNodesToSaveLocally = [NSMutableArray new];
+            NSMutableArray *arrayOfPathsForFilesToBeDeleted = [NSMutableArray new];
+            
+            if(node.isDocument)
             {
-                [self saveDeletedFileBeforeRemovingFromSync:(AlfrescoDocument *)node];
-                completionBlock(YES);
+                [weakSelf handleDocumentForDelete:node arrayOfNodesToDelete:arrayOfNodesToDelete arrayOfNodesToSaveLocally:arrayOfNodesToSaveLocally arrayOfPaths:arrayOfPathsForFilesToBeDeleted inRealm:backgroundRealm];
             }
-            else
+            else if(node.isFolder)
             {
-                [self.syncHelper deleteNodeFromSync:node inRealm:backgroundRealm];
-                completionBlock(NO);
+                [weakSelf handleFolderForDelete:node arrayOfNodesToDelete:arrayOfNodesToDelete arrayOfNodesToSaveLocally:arrayOfNodesToSaveLocally arrayOfPaths:arrayOfPathsForFilesToBeDeleted inRealm:backgroundRealm];
             }
-        }];
-    });
+            
+            BOOL hasSavedLocally = NO;
+            
+            if(arrayOfNodesToSaveLocally.count)
+            {
+                hasSavedLocally = YES;
+                for(AlfrescoDocument *document in arrayOfNodesToSaveLocally)
+                {
+                    [weakSelf saveDeletedFileBeforeRemovingFromSync:document];
+                }
+            }
+            
+            [[RealmManager sharedManager] deleteRealmObjects:arrayOfNodesToDelete inRealm:backgroundRealm];
+            for(NSString *path in arrayOfPathsForFilesToBeDeleted)
+            {
+                // No error handling here as we don't want to end up with Sync orphans
+                [weakSelf.fileManager removeItemAtPath:path error:nil];
+            }
+            
+            completionBlock(hasSavedLocally);
+        });
+    }
+}
+
+- (void)handleDocumentForDelete:(AlfrescoNode *)document arrayOfNodesToDelete:(NSMutableArray *)arrayToDelete arrayOfNodesToSaveLocally:(NSMutableArray *)arrayToSave arrayOfPaths:(NSMutableArray *)arrayOfPaths inRealm:(RLMRealm *)realm
+{
+    SyncNodeStatus *syncNodeStatus = [self.syncHelper syncNodeStatusObjectForNodeWithId:[self.syncHelper syncIdentifierForNode:document] inSyncNodesStatus:self.syncNodesStatus];
+    syncNodeStatus.totalSize = 0;
+    RealmSyncNodeInfo *nodeInfo = [[RealmManager sharedManager] syncNodeInfoForObjectWithId:[self.syncHelper syncIdentifierForNode:document] ifNotExistsCreateNew:NO inRealm:realm];
+    [arrayToDelete addObject:nodeInfo];
```

Comment:
There is a check being made in the FileFolderCollectionViewController that makes sure that this object will never be nil (it checks that the node is in the sync list, thus present in DB). Will add the check for nil here as well as a best practice.